### PR TITLE
Implement Origin UI in Traffic Portal

### PR DESCRIPTION
**Pull request by rawlinp**
Provide the ability to list, create, edit, and delete Origins using Traffic Portal.

### CODE REVIEW

**Comment by rawlinp** [comment]
This depends on #2314 because that PR changes the `isPrimary` field to read-only, and this PR does **not** send `isPrimary` in requests. Currently that field is required and updatable, but it shouldn't be. Hence the dependency on that PR.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1796/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1813/
Test PASSed.

File: traffic_portal/app/src/common/modules/form/origin/form.origin.tpl.html [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+        </ol>
+        <div class="clearfix"></div>
+    </div>
+    <div class="x_content">
+        <br>
+        <form name="originForm" class="form-horizontal form-label-left" novalidate>
+            <div ng-show='!settings.isNew' class="form-group">
+                <label class="control-label col-md-2 col-sm-2 col-xs-12">Primary:</label>
+                <div class="col-md-10 col-sm-10 col-xs-12">
+                    <input name="isPrimary" type="text" class="form-control" ng-value="origin.isPrimary" readonly autofocus>
```
Comment (mitchell852) [inline traffic_portal/app/src/common/modules/form/origin/form.origin.tpl.html]: is this a true|false? do you want to use a select instead?

Comment (rawlinp) [reply traffic_portal/app/src/common/modules/form/origin/form.origin.tpl.html]: `isPrimary` is a read-only field, so I just wanted TP to display it here without actually sending the field back in the PUT/POST request.

**Pull request merged**